script/weblatedl.go

```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2022 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+type stat struct {
+	Code       string `json:"code"`
+	Name       string `json:"name"`
+	Total      int    `json:"total"`
+	Translated int    `json:"translated"`
+	Fuzzy      int    `json:"fuzzy"`
+}
+
+type translation map[string]string
+
+func main() {
+	log.SetFlags(log.Lshortfile)
+
+	if t := authToken(); t == "" {
+		log.Fatal("Need environment variable WEBLATE_TOKEN")
+	}
+
+	curValidLangs := map[string]bool{}
+	for _, lang := range loadValidLangs() {
+		curValidLangs[lang] = true
+	}
+	log.Println(curValidLangs)
+
+	resp := req("https://hosted.weblate.org/exports/stats/syncthing/gui/?format=json")
+
+	var stats []stat
+	err := json.NewDecoder(resp.Body).Decode(&stats)
+	if err != nil {
+		log.Fatal(err)
+	}
+	resp.Body.Close()
+
+	names := make(map[string]string)
+
+	var langs []string
+	for _, stat := range stats {
+		code := reformatLanguageCode(stat.Code)
+		pct := 100 * stat.Translated / stat.Total
+		if pct < 75 || !curValidLangs[code] && pct < 95 {
+			log.Printf("Language %q too low completion ratio %d%%", code, pct)
```

Comment:
Why are you not continuing here?

Reply:
That's a difference between the fundamental data models in Weblate and Transifex. Weblate syncs from the upstream repo and expects any changed translations to get merged there at some point in time. Be it via automated pull requests (as configured right now on my branch), or via importing and committing the JSON via script. It then rebases any outstanding translation updates on top of the updated upstram branch.

If we skipped the rest of the loop here, incomplete languages would simply never get their changes integrated upstream. Regardless of completion state, the current JSON file should always be versioned upstream. The Transifex script actually deletes incomplete files, which would drop the languages from Weblate completely - not what we want.

Reply:
Ah yeah right. We just don't write it to that prettyprint.js file and thus it's ignored in the UI.